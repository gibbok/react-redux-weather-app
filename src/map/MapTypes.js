
// @flow
import * as React from 'react'
/* eslint-disable no-undef */
type PropsType = {
  +types: Array<{
    +id: string,
    +title: string,
    +isActive: boolean
  }>,
  +onTypesClick: Function
}
/* eslint-enable no-undef */

const MapTypes:React.StatelessFunctionalComponent<any> = ({ types, onTypesClick }:PropsType):React.Element<any> => {
  return (
    <div >
      {types.map((type) =>
        <div onClick={(e) => onTypesClick(type.id)}
          key={type.id}
          >{type.title}</div>)}
    </div>
  )
}

export default MapTypes
