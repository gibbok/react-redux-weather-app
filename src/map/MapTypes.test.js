import React from 'react'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import MapTypes from './MapTypes'

describe('<MapTypes />', () => {
  it('should render', () => {
    const data = {
      types: [
        {
          id: 'temperature',
          title: 'Temperature',
          isActive: true
        },
        {
          id: 'precipitation',
          title: 'Precipitation',
          isActive: false
        },
        {
          id: 'pressure',
          title: 'Pressure',
          isActive: false
        },
        {
          id: 'wind',
          title: 'Wind',
          isActive: false
        },
        {
          id: 'cloud',
          title: 'Cloud',
          isActive: false
        }
      ]
    }
    const wrapper = shallow(
      <MapTypes
        types={data.types}
      />
    )
    expect(toJson(wrapper)).toMatchSnapshot()
  })
})
