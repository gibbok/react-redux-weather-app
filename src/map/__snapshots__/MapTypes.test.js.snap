// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MapTypes /> should render 1`] = `
<div>
  <div
    onClick={[Function]}
  >
    Temperature
  </div>
  <div
    onClick={[Function]}
  >
    Precipitation
  </div>
  <div
    onClick={[Function]}
  >
    Pressure
  </div>
  <div
    onClick={[Function]}
  >
    Wind
  </div>
  <div
    onClick={[Function]}
  >
    Cloud
  </div>
</div>
`;
